agent-inject/agent/agent.go

```diff
@@ -272,6 +280,14 @@ func ShouldInject(pod *corev1.Pod) (bool, error) {
 func (a *Agent) Patch() ([]byte, error) {
 	var patches []byte
 
+	// Add our vault-approle-secrets volume
+	if a.AutoAuthMethod == "approle" {
```

Comment:
Would it be better to abstract this and have an interface that has two methods `GetAutoAuthPatches` and `GetAutoAuthConfig`